foundation-ui/mount/MountService.js

```diff
@@ -0,0 +1,86 @@
+import {EventEmitter} from 'events';
+import {CHANGE} from './MountEvent';
+
+/**
+ * List of component descriptors
+ * @type {Array.<{component:function, priority:Number, type:String}>}
+ */
+const components = [];
+
+/**
+ * MountService
+ */
+class MountService extends EventEmitter {
+
+  /**
+   * Register component with type and optional priority
+   *
+   * respective mounts
+   * @param {function} component - React component which will be
+   * mounted based on the provided type
+   * @param {String} type - String which is used to match components with the
+   * @param {Number} [priority] - Number which is used to sort the components
+   */
+  registerComponent(component, type, priority = 0) {
+    if (typeof component !== 'function') {
+      if (process.env.NODE_ENV !== 'production') {
+        throw new TypeError('Provided component must be a ' +
+            'React.Component constructor or a stateless functional component.');
+      }
+
+      return;
+    }
+
+    if (typeof type !== 'string' || type === '') {
+      if (process.env.NODE_ENV !== 'production') {
+        throw new TypeError('Provided type must be a none empty string');
+      }
+
+      return;
+    }
+
+    // Add component descriptor and sort components list by priority and index
+    components.push({component, priority, type, index: components.length});
+    components.sort((a, b) => {
+      if (a.priority !== b.priority) {
+        return b.priority - a.priority;
+      }
+
+      return a.index - b.index;
```

Comment:
I see a problem here:
* Register  a couple of components with varying priorities, call `unregisterComponent` -> Chance is you did not remove the item with the highest index. Calling `registerComponent` again will have unexpected behaviour since more than 1 items have the same `index`.

Suggestions:

1. You should also re-assign new `index` properties in the `unregisterComponent` function.
2. Use an ever-incrementing value for index, instead of `components.length`